File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <iostream>
#include <unordered_map>


using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    unordered_map<string, int> name;
    for(int i=0; i<participant.size(); i++){
        name[participant[i]]++;
    }
    
    for(int i=0; i<completion.size(); i++){
        name[completion[i]]--;
    }
    
    for(auto pair : name){
        if(pair.second > 0)
            answer = pair.first;
    }
    
    cout << answer;
    
    return answer;
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.HashMap;

class Solution {
    public String solution(String[] participant, String[] completion) {
        HashMap<String, Integer> players = new HashMap<>();
        
        for (String player : completion) {
            players.put(player, players.getOrDefault(player, 0) + 1);
        }
        
        for (String s : participant) {
            if (players.getOrDefault(s, 0) == 0) {
                return s;
            }
            players.put(s, players.get(s) - 1);
        }
        
        return null;
    }
}